crates/cext/src/param.rs

```diff
@@ -0,0 +1,1120 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+use num_complex::Complex64;
+use std::ffi::{c_char, CStr, CString};
+use std::sync::Arc;
+
+use crate::exit_codes::ExitCode;
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::operations::Param;
+use qiskit_circuit::parameter::parameter_expression::{ParameterError, ParameterExpression};
+use qiskit_circuit::parameter::symbol_expr::{Symbol, SymbolExpr, Value};
+
+/// @ingroup QkParam
+/// Construct a new Parameter with a name of symbol
+///
+/// @param name The name of symbol.
+///
+/// @return A pointer to the created QkParam.
+///
+/// # Example
+///
+///     QkParam *a = qk_param_new("a");
+///
+/// # Safety
+///
+/// name must be a pointer to valid C string to name of symbol
```

Comment:
We should make sure that these are all proper sentences 🙂 You could e.g. check existing safety comments to ensure they are correct & consistent. E.g. from the `qk_quantum_register_new` function:
```suggestion
/// The `name` parameter must be a pointer to memory that contains a valid
/// nul terminator at the end of the string. It also must be valid for reads of
/// bytes up to and including the nul terminator.
```